src/menu/SongItem.as

```diff
@@ -1,131 +1,254 @@
     import classes.Text;
     import flash.display.Sprite;
+    import flash.display.GradientType;
+    import flash.geom.Matrix;
+    import flash.text.StyleSheet;
     import flash.events.MouseEvent;
+    import com.flashfla.utils.sprintf;
+    import com.flashfla.utils.NumberUtil;
+    import classes.Language;
+    import flash.net.navigateToURL;
+    import flash.net.URLRequest;
+    import flash.text.TextField;
+    import flash.text.AntiAliasType;
+    import flash.text.TextFieldAutoSize;
+    import flash.ui.ContextMenu;
+    import flash.events.Event;
 
     public class SongItem extends Sprite
     {
-        private var _active:Boolean = false;
-        private var _song:Array;
-        private var _rank:Object;
+        /** Marks the Button as in-use to avoid removal in song selector. */
+        //public var garbageSweep:Boolean = false;
 
-        //- Song Details
-        public var level:Number;
-        public var genre:Number;
-        public var index:Number;
-        public var box:Box;
+        /** Calculated y position in the scroll pane. */
+        //public var fixed_y:Number = 0;
 
-        private var nameText:Text;
-        private var iconText:Text;
-        private var diffText:Text;
+        public var index:int = 0;
 
-        public function SongItem(sO:Array, rO:Object, isActive:Boolean = false):void
-        {
-            this._active = isActive;
-            this._song = sO;
-            this._rank = rO;
+        // Text
+        private var _lblSongDifficulty:Text;
+        private var _lblSongName:Text;
+        private var _lblSongFlag:Text;
 
-            buildBox();
+        private var _lblMessageText:TextField;
 
+        // Display
+        private var _width:Number = 400;
+        private var _height:Number = 27;
+        private var _highlight:Boolean = false;
+        private var _active:Boolean = false;
+
+        // Song Data
+        private var _songData:Object;
+        private var _level:int = 0;
+        public var isLocked:Boolean = false;
+
+        public function SongItem()
+        {
             //- Set Button Mode
             this.mouseChildren = false;
             this.useHandCursor = true;
             this.buttonMode = true;
 
-            this.addEventListener(MouseEvent.ROLL_OVER, boxOver);
+            //- Events
+            this.addEventListener(MouseEvent.ROLL_OVER, e_onHover);
         }
 
-        public function dispose():void
+        public function draw():void
         {
-            this.removeEventListener(MouseEvent.ROLL_OVER, boxOver);
-            this.removeEventListener(MouseEvent.ROLL_OUT, boxOut);
-
-            //- Remove is already existed.
-            if (box != null)
+            this.graphics.clear();
+            this.graphics.lineStyle(1, 0xFFFFFF, (highlight ? 0.8 : 0.55));
+            this.graphics.beginGradientFill(GradientType.LINEAR, [0xFFFFFF, 0xFFFFFF], (highlight ? [0.35, 0.1225] : [0.2, 0.04]), [0, 255], Constant.GRADIENT_MATRIX);
```

Comment:
Lines should not run off the page, please separate highlight tertiary to separate line.